lib/puppet-strings/yard/handlers/ruby/function_handler.rb

```diff
@@ -133,36 +134,40 @@ def add_overload_tag(object, node)
       method_name = child.method_name.source
       next unless DISPATCH_METHOD_NAMES.include?(method_name)
 
-      # Check for block
-      if method_name.include?('block')
-        if block
-          log.warn "A duplicate block parameter was found for Puppet function '#{object.name}' at #{child.file}:#{child.line}."
+      if method_name == 'return_type'
+        overload_tag.tag(:return).types = [node_as_string(child.parameters[0])]
+      else
```

Comment:
could just do a `next` to get out of this conditional. Then there is no need to nest the rest of the logic under this conditional's `else` (right now there is a construct `else` `if`` which should really be an `elseif` - but since all other cases use `next` this case ('return_type') could use that as well as it is easier to read without the nesting.